src/librustc_resolve/resolve_imports.rs

```diff
@@ -1238,17 +1239,27 @@ impl<'a, 'b:'a> ImportResolver<'a, 'b> {
             if is_good_import || binding.is_macro_def() {
                 let def = binding.def();
                 if def != Def::Err {
-                    if let Some(def_id) = def.opt_def_id() {
-                        if !def_id.is_local() && def_id.krate != CrateNum::BuiltinMacros {
-                            self.cstore.export_macros_untracked(def_id.krate);
+                    let exp = if let NameBindingKind::Import { directive , .. } = binding.kind {
+                        match directive.subclass {
+                            ImportDirectiveSubclass::ExternCrate { no_link: true, .. } => false,
+                            _ => true
                         }
+                    } else {
+                        true
+                    };
```

Comment:
Could all of these be part of the `impl` of `ImportDirectiveSubclass` or `NameBindingKind`?

Reply:
These changes are a leftover hack which should be removed now.